massa-module-cache/src/hd_cache.rs

```diff
@@ -58,9 +58,22 @@ impl HDCache {
     /// * path: where to store the db
     /// * max_entry_count: maximum number of entries we want to keep in the db
     /// * amount_to_remove: how many entries are removed when `entry_count` reaches `max_entry_count`
+    ///
+    /// Note: entry_count refers to the number of modules stored in the db, which is half the number of (key, value) pairs in the db
     pub fn new(path: PathBuf, max_entry_count: usize, snip_amount: usize) -> Self {
-        let db = DB::open_default(path).expect(OPEN_ERROR);
-        let entry_count = db.iterator(IteratorMode::Start).count();
+        let mut db = DB::open_default(path.clone()).expect(OPEN_ERROR);
+        let mut nb_db_entries = db.iterator(IteratorMode::Start).count();
+        if nb_db_entries % 2 != 0 {
```

Comment:
maybe do with it with checked_div ?

Reply:
checked_div does not check for divisibility, only for divisions by 0